app/src/fdroid/java/org/fossasia/openevent/general/search/location/LocationServiceImpl.kt

```diff
@@ -61,4 +61,47 @@ class LocationServiceImpl(
             }
         }
     }
+
+    @SuppressLint("MissingPermission")
+    override fun getCityName(): Single<String> {
+        val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as? LocationManager
+            ?: return Single.error(IllegalStateException())
+        val geoCoder = Geocoder(context, Locale.getDefault())
+
+        return Single.create { emitter ->
+            try {
+                locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 0f,
+                    object : LocationListener {
+                        override fun onLocationChanged(location: Location?) {
+                            if (location == null) {
+                                emitter.onError(java.lang.IllegalStateException())
```

Comment:
What IllegalState?

Reply:
Given for the condition if incase it fails to update location to the variable then it gives the error for it

Reply:
I can't see that message in IllegalStateException constructor. How would I know?

Reply:
should I remove the it? or I had a question is there need of AdminArea if we are implementing it using cityName if not then should I change the method for calling the admin area with the city name as it will also save from creating another function for it?

Reply:
No, you should throw error with a proper message

Reply:
okay,  I will fix it soon